pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -145,6 +146,19 @@ public String consumerName() {
         return consumerName;
     }
 
+    void notifyConsumerGroupChange(long activeConsumerId) {
+        if (cnx.getRemoteEndpointProtocolVersion() < ProtocolVersion.v11.getNumber()) {
+            // if the client is older than `v11`, we don't need to send consumer group changes.
+            return;
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("notify consumer {} - that [{}] for subscription {} has new active consumer : {}",
+                consumerId, topicName, subscription.getName(), activeConsumerId);
+        }
+        cnx.ctx().write(Commands.newConsumerGroupChange(consumerId, activeConsumerId));
```

Comment:
Use `writeAndFlush()`, otherwise the message won't be written on the socket until and unrelated flush is triggered.

Also, pass `cnx.ctx().voidPromise()` as the 2nd parameter to avoid creating the write promise since we don't care here.